src/app/containers/PageHandlers/withVariant/index.jsx

```diff
@@ -1,50 +1,41 @@
-import PropTypes from 'prop-types';
-import pathOr from 'ramda/src/pathOr';
 import React from 'react';
 import { useParams, useLocation, Redirect } from 'react-router-dom';
-import getVariantRedirectUrl from './getVariantRedirectUrl';
+import { variantSanitiser, getVariant } from '#lib/utilities/variantHandler';
+import {
+  getPreferredVariant,
+  setPreferredVariant,
+} from '#contexts/UserContext/cookies';
+
+const redirect = (location, service, defaultVariant) => {
+  const preferredVariant = getPreferredVariant(service);
+  return (
+    <Redirect
+      to={{
+        ...location,
+        pathname: `${location.pathname}/${preferredVariant || defaultVariant}`,
+      }}
+    />
+  );
+};
 
-const WithVariant = Component => {
+const withVariant = Component => {
   const VariantContainer = props => {
```

Comment:
Can we just return an anonymous function instead? I'm not sure what is container-y about a component returned in a HOC

```suggestion
  return props => {
```

Reply:
Can we still change this please?